app/models/deploy_service.rb

```diff
@@ -32,16 +32,13 @@ def auto_confirm?(stage)
   end
 
   def latest_approved_deploy(reference, project)
-    deploy = nil
-    Deploy.where(reference: reference).where(' buddy_id is NOT NULL AND started_at > ?', BuddyCheck.period.hours.ago)
+    Deploy.where(reference: reference).where('buddy_id is NOT NULL AND started_at > ?', BuddyCheck.period.hours.ago)
       .includes(:stage)
-      .where(stages: {project_id: project, production: true})
+      .where(stages: {project_id: project})
       .each do |d|
```

Comment:
Ah, I missed that comment.  Yeah, `detect` would be better here.  I'll create another PR to clean up this class a little bit.